user.go

```diff
@@ -108,53 +157,109 @@ func (s *Server) handlerUser(w http.ResponseWriter, r *http.Request) {
+					data.ErrCurrentPass = true
+					data.PassError = append(data.PassError, "Invalid current password")
+				}
 
-			if newPass1_raw != newPass2_raw {
-				data.ErrNewPass = true
-				data.PassError = append(data.PassError, "Passwords do not match")
-			}
+				if newPass1_raw == "" {
+					data.ErrNewPass = true
+					data.PassError = append(data.PassError, "New password cannot be blank")
+				}
 
-			if !(data.ErrCurrentPass || data.ErrNewPass || data.ErrEmail) {
-				// Change pass
-				data.SuccessMessage = "Password successfully changed"
-				user.Password = s.hashPassword(newPass1_raw)
-				user.PassDate = time.Now()
+				if newPass1_raw != newPass2_raw {
+					data.ErrNewPass = true
+					data.PassError = append(data.PassError, "Passwords do not match")
+				}
+				if !(data.ErrCurrentPass || data.ErrNewPass || data.ErrEmail) {
+					// Change pass
+					data.SuccessMessage = "Password successfully changed"
+					localAuth.Password = s.hashPassword(newPass1_raw)
+					localAuth.PassDate = time.Now()
+
+					if err = s.data.UpdateAuthMethod(localAuth); err != nil {
+						s.l.Error("Unable to save User with new password:", err)
+						s.doError(http.StatusInternalServerError, "Unable to update password", w, r)
+						return
+					}
+
+					s.l.Info("new PassDate: %s", localAuth.PassDate)
+					err = s.login(user, common.AUTH_LOCAL, w, r)
+					if err != nil {
+						s.l.Error("Unable to login to session:", err)
+						s.doError(http.StatusInternalServerError, "Unable to update password", w, r)
+						return
+					}
+				}
+			}
+		} else if formVal == "Notifications" {
+			// Update notifications
+		} else if formVal == "SetPassword" {
+			pass1_raw := r.PostFormValue("Password1")
+			pass2_raw := r.PostFormValue("Password2")
 
-				s.l.Info("new PassDate: %s", user.PassDate)
+			_, err := user.GetAuthMethod(common.AUTH_LOCAL)
+			if err == nil {
+				data.ErrCurrentPass = true
+				data.PassError = append(data.PassError, "Existing password detected. (how did you end up here anyways?)")
+			} else {
+				localAuth := &common.AuthMethod{
+					Type: common.AUTH_LOCAL,
+				}
 
-				err = s.login(user, w, r)
-				if err != nil {
-					s.l.Error("Unable to login to session:", err)
-					s.doError(http.StatusInternalServerError, "Unable to update password", w, r)
-					return
+				if pass1_raw == "" {
+					data.ErrNewPass = true
+					data.PassError = append(data.PassError, "New password cannot be blank")
 				}
 
-				if err = s.data.UpdateUser(user); err != nil {
-					s.l.Error("Unable to save User with new password:", err)
-					s.doError(http.StatusInternalServerError, "Unable to update password", w, r)
-					return
+				if pass1_raw != pass2_raw {
+					data.ErrNewPass = true
+					data.PassError = append(data.PassError, "Passwords do not match")
+				}
+				if !(data.ErrCurrentPass || data.ErrNewPass || data.ErrEmail) {
+					// Change pass
+					data.SuccessMessage = "Password successfully changed"
```

Comment:
This success message is wrong, the user didnt chane the password, he set a new password.

Reply:
fixed in a06a620